drivers/serial/uart_pl011.c

```diff
@@ -125,6 +125,10 @@ struct pl011_data {
 #define PL011_CR_UARTEN		BIT(0)	/* enable uart operations */
 #define PL011_CR_SIREN		BIT(1)	/* enable IrDA SIR */
 #define PL011_CR_SIRLP		BIT(2)	/* IrDA SIR low power mode */
+#ifdef CONFIG_UART_PL011_AMBIQ_CLK_CR_REG
+#define PL011_CR_CLKEN		BIT(3)	/* clock enable */
+#define PL011_CR_CLKSEL		GENMASK(6, 4) /* clock select */
+#endif
```

Comment:
I'm not the UART maintainer, but IMO vendor quirks on generic IP drivers should be split in a separate file.

Reply:
I've added support for vendor quirks in the PL011 driver, and moved Ambiq-specific quirks to a separate file.